domains/transfer/Transfer.API/API.IntegrationTests/Migrations/AddUniqueIndexAndTransferAgreementNumberTests.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using API.Data;
+using API.IntegrationTests.Factories;
+using API.Models;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.Extensions.DependencyInjection;
+using Npgsql;
+using Xunit;
+
+namespace API.IntegrationTests.Migrations;
+
+public class AddUniqueIndexAndTransferAgreementNumberTests : IClassFixture<TransferAgreementsApiWebApplicationFactory>
+{
+    private readonly TransferAgreementsApiWebApplicationFactory factory;
+
+    public AddUniqueIndexAndTransferAgreementNumberTests(TransferAgreementsApiWebApplicationFactory factory)
+    {
+        this.factory = factory;
+    }
+
+    [Fact]
+    public async Task ApplyMigration_WhenExistingDataInDatabase_Success()
+    {
+        var dbContextFactory = factory.Services.GetRequiredService<IDbContextFactory<ApplicationDbContext>>();
+        await using var dbContext = await dbContextFactory.CreateDbContextAsync();
+
+        var migrator = dbContext.Database.GetService<IMigrator>();
+
+        await migrator.MigrateAsync("20230829090644_AddInvitationsTable");
+        await dbContext.TruncateTransferAgreementsTable();
+
+        await InsertOldTransferAgreement(dbContext, new TransferAgreement
+        {
+            EndDate = DateTimeOffset.UtcNow.AddDays(1),
+            Id = Guid.NewGuid(),
+            ReceiverReference = Guid.NewGuid(),
+            ReceiverTin = "11223344",
+            SenderId = Guid.NewGuid(),
+            SenderName = "Producent A/S",
+            SenderTin = "12345678",
+            StartDate = DateTimeOffset.UtcNow
+        });
+        await InsertOldTransferAgreement(dbContext, new TransferAgreement
+        {
+            EndDate = DateTimeOffset.UtcNow.AddDays(1),
+            Id = Guid.NewGuid(),
+            ReceiverReference = Guid.NewGuid(),
+            ReceiverTin = "11223345",
+            SenderId = Guid.NewGuid(),
+            SenderName = "Producent A/S2",
+            SenderTin = "12345679",
+            StartDate = DateTimeOffset.UtcNow
+        });
+
+        var applyMigration = () => migrator.Migrate("20230829124003_AddUniqueIndexAndTransferAgreementNumber");
+        applyMigration.Should().NotThrow();
+
+        var tas = dbContext.TransferAgreements.ToList();
+
+        tas.Count.Should().Be(2);
+
+    }
+
+    [Fact]
+    public async Task ApplyMigration_WhenTransferAgreementsHasSameSenderId_ExpectError()
+    {
+        var dbContextFactory = factory.Services.GetRequiredService<IDbContextFactory<ApplicationDbContext>>();
+        await using var dbContext = await dbContextFactory.CreateDbContextAsync();
+
+        var migrator = dbContext.Database.GetService<IMigrator>();
+
+        await migrator.MigrateAsync("20230829090644_AddInvitationsTable");
```

Comment:
Depending on the order of the tests, this could actually be a "down" migration and not an up. This is due to not using a clean database for each test (or cleaning/dropping your entire database between tests).

Reply:
Yes but what is the problem? I migrate to the point i need